python/dgl/distributed/partition.py

```diff
@@ -1410,8 +1554,21 @@ def init_type_per_edge(graph, gpb):
     return etype_ids
 
 
+def _load_parts(part_config, part_id, parts):
+    """load parts from variable or dist."""
+    if parts is None:
+        graph, _, _, _, _, _, _ = load_partition(
+            part_config, part_id, load_feats=False
+        )
+    else:
+        graph = parts[part_id]
+    return graph
+
+
 def gb_convert_single_dgl_partition(
     part_id,
+    parts,
+    part_meta,
```

Comment:
passing `part_meta`, `part_config`, `parts` to a function looks a bit mixing and confusing. how about split `gb_convert_single_dgl_partition` into pieces such as `read_partition_from_file() if needed`, `prepare metadata like gpb, ntypes, etypes`, `convert loaded partition to target format given metadata`. just split into smaller functions that are clearer. Adhere to the Single Responsibility Principle for each function.